util/json.go

```diff
@@ -19,3 +25,35 @@ func JsonReMarshalString(s string) (string, error) {
 	}
 	return string(output), nil
 }
+
+func ReadJSON(r *http.Request, dst interface{}) error {
+	var syntaxError *json.SyntaxError
+	var unmarshalTypeError *json.UnmarshalTypeError
+	var invalidUnMarshalError *json.InvalidUnmarshalError
+
+	err := json.NewDecoder(r.Body).Decode(dst)
+	if err != nil {
+		switch {
+		case errors.As(err, &syntaxError):
+			return fmt.Errorf("body contains badly-formed JSON (at character %d)", syntaxError.Offset)
+		case errors.Is(err, io.ErrUnexpectedEOF):
+			return errors.New("body contains badly-formed JSON")
+		case errors.As(err, &unmarshalTypeError):
+			if unmarshalTypeError.Field != "" {
+				return fmt.Errorf("body contains incorrect JSON type for field %q", unmarshalTypeError.Field)
+			}
+			return fmt.Errorf("body contains incorrect JSON type (at character %d)", unmarshalTypeError.Offset)
+
+		case errors.Is(err, io.EOF):
+			return errors.New("body must not be empty")
+
+		case errors.As(err, &invalidUnMarshalError):
+			panic(err)
```

Comment:
Do we have a global panic recoverer? i thought it was bad practice to panic

Reply:
I think we do, but we shouldn't panic like this. @Dotunj 👀

Reply:
I don't think it's necessarily a bad thing to panic at this stage. If we get the` invalidUnMarshalError` at runtime, it is most likely an error on the developer's part indicating that the wrong type was passed to `Decode()`